core/vm/vmcontext/context.go

```diff
@@ -142,25 +142,30 @@ func VerifySealFn(ref *types.Header, chain chainContext) func(*types.Header) boo
 // If the calculation or transfer of the tax amount fails for any reason, the regular transfer goes ahead.
 // NB: Gas is not charged or accounted for this calculation.
 func TobinTransfer(evm *vm.EVM, sender, recipient common.Address, amount *big.Int) {
-	// Run only primary evm.Call() with tracer
-	if evm.GetDebug() {
-		evm.SetDebug(false)
-		defer func() { evm.SetDebug(true) }()
-	}
+	// Only deduct tobin tax before the g hardfork
+	if evm.ChainConfig().IsGFork(evm.Context.BlockNumber) {
+		Transfer(evm.StateDB, sender, recipient, amount)
+	} else {
+		// Run only primary evm.Call() with tracer
+		if evm.GetDebug() {
```

Comment:
Mh, looking at this again, didn't this code before disable tracing for the expected part of the transfer?

Reply:
It did disable it for the whole transfer but I think this is practically the same as what's happening here ^ since `Transfer` doesn't actually make any EVM/interpreter calls. It may be easier to reason about if you keep this on the outside of the hardfork logic anyways (should be functionally the same); also, this may be moved directly into the `SharedEVMRunner` anyways as part of this ticket https://github.com/celo-org/celo-blockchain/issues/2078